blockcypher/api.py

```diff
@@ -1499,6 +1500,15 @@ def create_unsigned_tx(inputs, outputs, change_address=None,
             inputs_cleaned.append({
                 'addresses': [address, ],
                 })
+        elif 'prev_hash' and  'output_index' in input_obj:
+            prev_hash = input_obj['prev_hash']
+            output_index = input_obj['output_index']
+            assert uses_only_hash_chars(prev_hash), prev_hash
+            assert str(output_index).isdigit(), output_index
```

Comment:
Clever construction, I hadn't seen this before.

I don't love that we're converting an int to a string in order to check that it's a non-negative int.

I think `assert type(output_index) is int, output_index` is much clearer but two immediate problems come to mind:
- For python2 it would be possible for data to be represented as a `Long`, so we'd have to check for that as well. Yuck.
- This doesn't check the case of `output_index < 0` (your check does). Not the end of the world as the API should catch that anyway and return an error.

tl:dr; Lot of rambling. I don't have a better suggestion than what you've already done!